cmd/bosun/web/web.go

```diff
@@ -119,6 +124,25 @@ func Listen(listenAddr string, devMode bool, tsdbHost string) error {
 	router.Handle("/api/tagv/{tagk}/{metric}", JSON(TagValuesByMetricTagKey))
 	router.Handle("/api/tagsets/{metric}", JSON(FilteredTagsetsByMetric))
 	router.Handle("/api/opentsdb/version", JSON(OpenTSDBVersion))
+	router.Handle("/api/annotate", JSON(AnnotateEnabled))
+
+	// Annotations
+	if schedule.Conf.AnnotateEnabled() {
+		var err error
+		index := schedule.Conf.AnnotateIndex
+		if index == "" {
+			index = "annotate"
+		}
+		annotateBackend, err = backend.NewElastic(schedule.Conf.AnnotateElasticHosts, index)
+		if err != nil {
+			return err
+		}
+		if err := annotateBackend.InitBackend(); err != nil {
+			return err
+		}
+		web.AddRoutes(router, "/api", []backend.Backend{annotateBackend}, false, false)
```

Comment:
enable the independent UI, and use local dev files when doing so

On Mon, Feb 22, 2016 at 11:56 AM, Craig Peterson <notifications@github.com>
wrote:

> In cmd/bosun/web/web.go
> <https://github.com/bosun-monitor/bosun/pull/1610#discussion_r53654814>:
>
> > +
> > +	// Annotations
> > +	if schedule.Conf.AnnotateEnabled() {
> > +		var err error
> > +		index := schedule.Conf.AnnotateIndex
> > +		if index == "" {
> > +			index = "annotate"
> > +		}
> > +		annotateBackend, err = backend.NewElastic(schedule.Conf.AnnotateElasticHosts, index)
> > +		if err != nil {
> > +			return err
> > +		}
> > +		if err := annotateBackend.InitBackend(); err != nil {
> > +			return err
> > +		}
> > +		web.AddRoutes(router, "/api", []backend.Backend{annotateBackend}, false, false)
>
> I like this way of doing it a lot. What are the last two bool parameters
> for? Mostly out of curiosity.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/bosun-monitor/bosun/pull/1610/files#r53654814>.
>